src/Handler/Table/Export/ExportTableToFileHandler.php

```diff
@@ -0,0 +1,153 @@
+        $fileCredentials = $any->unpack();
+        assert(
+            $fileCredentials instanceof S3Credentials,
+            'TableExportToFileCommand.fileCredentials is required to be S3Credentials.'
+        );
+
+        $commandExportOptions = $command->getExportOptions();
+
+        $commandExportOptionsTeradata = null;
+        if ($commandExportOptions) {
+            $commandExportOptionsTeradata = MetaHelper::getMetaFromCommand($commandExportOptions, TeradataExportOptionsMeta::class);
+            assert(
+                $commandExportOptionsTeradata instanceof TeradataExportOptionsMeta,
+                'TableExportToFileCommand.exportOptions.meta is required to be TeradataExportOptionsMeta.'
+            );
+        }
+
+        $exportOptions = $this->createOptions(
+            $commandExportOptions,
+            $credentials,
+            $commandExportOptionsTeradata
+        );
+
+
+        // run
+        $db = $this->manager->createSession($credentials);
+
+        $sourceRef = new Table(
+            ProtobufHelper::repeatedStringToArray($source->getPath())[0],
+            $source->getTableName()
+        );
+
+        $destinationRef = $this->getDestinationFile($command->getFilePath(), $fileCredentials);
+
+        (new Exporter($db))->exportTable(
+            $sourceRef,
+            $destinationRef,
+            $exportOptions
+        );
+
+        // TODO jaky bude obsah response?
+        return new TableExportResponse();
+    }
+
+    private function getDestinationFile(
+        FilePath $filePath,
+        S3Credentials $fileCredentials
+    ): DestinationFile {
+        $relativePath = RelativePath::create(
+            new S3Provider(),
+            $filePath->getRoot(),
+            $filePath->getPath(),
+            $filePath->getFileName(),
+        );
+        return new DestinationFile(
+            $fileCredentials->getKey(),
+            $fileCredentials->getSecret(),
+            $fileCredentials->getRegion(),
+            $relativePath->getRoot(),
+            $relativePath->getPathnameWithoutRoot()
+        );
+    }
+
+    private function createOptions(
+        ExportOptions $options,
+        GenericBackendCredentials $credentials,
+        ?TeradataExportOptionsMeta $meta
+    ): TeradataExportOptions {
+        return new TeradataExportOptions(
+            $credentials->getHost(),
+            $credentials->getPrincipal(),
+            $credentials->getSecret(),
+            $credentials->getPort(),
+            $options->getIsCompressed(),
+            $meta && $meta->getBufferSize() ? $meta->getBufferSize() : TeradataExportOptions::DEFAULT_BUFFER_SIZE,
+            $meta && $meta->getMaxObjectSize() ? $meta->getMaxObjectSize() : TeradataExportOptions::DEFAULT_MAX_OBJECT_SIZE,
+            // TODO pouzit default TeradataExportOptions::DEFAULT_SPLIT_ROWS = true?
+            $meta && $meta->getDontSplitRows() ? $meta->getDontSplitRows() : false,
```

Comment:
Tady je to docela matouci, protoze pokud neposlu zadnou hodnotu, tak ProtoBuffer vrati FALSE, ale tady je default na TRUE 🤯  Nejaky napad?

Reply:
vyhozeno, ponechan default.